benchmarks/circuit_construction.py

```diff
@@ -0,0 +1,110 @@
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Set
+
+import cirq
+
+
+def get_qubits(distance: int):
+    data_qubits: Set[cirq.Qid] = set()
+    z_measure_qubits: Set[cirq.Qid] = set()
+    x_measure_qubits: Set[cirq.Qid] = set()
+    for x in range(distance + 1):
+        for y in range(distance + 1):
+            if x < distance and y < distance:
+                data_qubits.add(cirq.GridQubit(2 * x + 1, 2 * y + 1))
+            on_boundary_x = (x == 0) or (x == distance)
+            on_boundary_y = (y == 0) or (y == distance)
+            parity = x % 2 != y % 2
+            if (on_boundary_x and parity) or (on_boundary_y and not parity):
+                continue
+            if parity:
+                z_measure_qubits.add(cirq.GridQubit(2 * x, 2 * y))
+            else:
+                x_measure_qubits.add(cirq.GridQubit(2 * x, 2 * y))
+    return data_qubits, z_measure_qubits, x_measure_qubits
+
+
+def rotated_surface_code_memory_z_cycle(
+    data_qubits, z_measure_qubits, x_measure_qubits, z_order, x_order
+) -> cirq.OP_TREE:
+    yield cirq.Moment([cirq.H(q) for q in x_measure_qubits])
+    for k in range(4):
+        op_list = []
+        for measure_qubits, add, is_x in [
+            [x_measure_qubits, x_order[k], True],
+            [z_measure_qubits, z_order[k], False],
+        ]:
+            for q_meas in measure_qubits:
+                q_data = q_meas + add
+                if q_data in data_qubits:
+                    op_list.append(cirq.CNOT(q_meas, q_data) if is_x else cirq.CNOT(q_data, q_meas))
+        yield cirq.Moment(op_list)
+    yield cirq.Moment([cirq.H(q) for q in x_measure_qubits])
+    yield cirq.Moment(cirq.measure_each(*x_measure_qubits, *z_measure_qubits))
+
+
+def surface_code_circuit(
+    distance: int, num_rounds: int, moment_by_moment: bool = True
+) -> cirq.Circuit:
+    data_qubits, z_measure_qubits, x_measure_qubits = get_qubits(distance)
+    x_order = [(1, 1), (1, -1), (-1, 1), (-1, -1)]
+    z_order = [(1, 1), (-1, 1), (1, -1), (-1, -1)]
+    surface_code_cycle = cirq.Circuit(
+        rotated_surface_code_memory_z_cycle(
+            data_qubits, x_measure_qubits, z_measure_qubits, x_order, z_order
+        )
+    )
+    if moment_by_moment:
+        return cirq.Circuit(
+            surface_code_cycle * num_rounds, cirq.Moment(cirq.measure_each(*data_qubits))
+        )
+    else:
+        return cirq.Circuit(
+            [*surface_code_cycle.all_operations()] * num_rounds, cirq.measure_each(*data_qubits)
+        )
+
+
+class SurfaceCode_Rotated_Memory_Z:
```

Comment:
Prefer `SurfaceCodeRotatedMemoryZ` - all other Cirq classes are pure CamelCase, not a camel-snake hybrid.

Reply:
done